lib/saluki-components/src/transforms/name_filter/mod.rs

```diff
@@ -0,0 +1,234 @@
+use std::ops::Deref;
+
+use async_trait::async_trait;
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_config::GenericConfiguration;
+use saluki_core::{
+    components::{
+        transforms::{Transform, TransformBuilder, TransformContext},
+        ComponentContext,
+    },
+    topology::OutputDefinition,
+};
+use saluki_error::GenericError;
+use saluki_event::{metric::Metric, DataType};
+use serde::Deserialize;
+use tokio::select;
+use tracing::{debug, error};
+
+#[derive(Deserialize)]
+/// Name Transform.
+pub struct NameFilterConfiguration {
+    #[serde(default, rename = "statsd_metric_namespace")]
+    metric_prefix: String,
+
+    #[serde(
+        default = "default_metric_prefix_blacklist",
+        rename = "statsd_metric_namespace_blacklist"
+    )]
+    metric_prefix_blacklist: Vec<String>,
+
+    #[serde(default, rename = "statsd_metric_blocklist")]
+    metric_blocklist: Vec<String>,
+
+    #[serde(default, rename = "statsd_metric_blocklist_match_prefix")]
+    metric_blocklist_match_prefix: bool,
+}
+
+fn default_metric_prefix_blacklist() -> Vec<String> {
+    vec![
+        "datadog.agent".to_string(),
+        "datadog.dogstatsd".to_string(),
+        "datadog.process".to_string(),
+        "datadog.trace_agent".to_string(),
+        "datadog.tracer".to_string(),
+        "activemq".to_string(),
+        "activemq_58".to_string(),
+        "airflow".to_string(),
+        "cassandra".to_string(),
+        "confluent".to_string(),
+        "hazelcast".to_string(),
+        "hive".to_string(),
+        "ignite".to_string(),
+        "jboss".to_string(),
+        "jvm".to_string(),
+        "kafka".to_string(),
+        "presto".to_string(),
+        "sidekiq".to_string(),
+        "solr".to_string(),
+        "tomcat".to_string(),
+        "runtime".to_string(),
+    ]
+}
+
+impl NameFilterConfiguration {
```

Comment:
We should probably call this like... `DogStatsDPrefixFilterConfiguration` (and `DogStatsDPrefixFilter` for the component itself) since `NameFilter` doesn't _really_ explain what it's doing (certainly not that it might be prefixing the metrics, at least.)